flight/UAVTalk/uavtalk.c

```diff
@@ -391,7 +377,7 @@ UAVTalkRxState UAVTalkProcessInputStreamQuiet(UAVTalkConnection connectionHandle
 		}
 
 		// Check the lengths match
-		if ((iproc->rxPacketLength + iproc->instanceLength + iproc->timestampLength + iproc->length) != iproc->packet_size) { // packet error - mismatched packet size
+		if ((iproc->rxPacketLength + iproc->instanceLength + iproc->length) != iproc->packet_size) { // packet error - mismatched packet size
```

Comment:
> There exist a number of 3rd party implementations in the wild and they are all free to send timestamped packets of any type (perfectly valid under the protocol spec)

If there's any non-logging implementation that sends these timestamps I would be more interested.

I don't want to be caught with design mistakes made 5 years ago forever.  We can change uavtalk.  There's a big advantage from simplifying this down, both in message bits we're reclaiming for possible future use (though this has potential caveats and has to be handled carefully), simplifying state machines, and enabling further simplification.

"Just for a few lines of code" also happened to fix a severe bug that you've fixed in another PR, for instance.

> That's the point, if you make a BC change you should also change the version.

You should change the version if you do something that will cause information from past revisions to be misinterpreted.

With the protocol version change: all messages are rejected from old implementations.

Without the protocol version change: messages we do not believe sent anywhere are rejected.

There are far bigger changes we've made with interoperability considerations.  E.g. the way some things are quasi-pipelined now that were serialized before.  The new implementation (as of a few months ago) is still compliant with the specification, but broke interop with most old implementations.